commit e5b9f4b1dc3e73f8a327b242c783551c4bf15bbd
Author: Gustavo A. R. Silva <gustavoars@kernel.org>
Date:   Thu Jul 23 17:31:27 2020 -0500

    tty: Avoid the use of one-element arrays
    
    One-element arrays are being deprecated[1]. Replace the one-element arrays
    with simple value types 'char reserved_char' and 'compat_int_t reserved'[2],
    once it seems these are just placeholders for alignment.
    
    [1] https://github.com/KSPP/linux/issues/79
    [2] https://github.com/KSPP/linux/issues/86
    
    Tested-by: kernel test robot <lkp@intel.com>
    Link: https://github.com/GustavoARSilva/linux-hardening/blob/master/cii/0-day/tty-20200716.md
    Acked-by: Jiri Slaby <jirislaby@kernel.org>
    Signed-off-by: Gustavo A. R. Silva <gustavoars@kernel.org>
    Link: https://lore.kernel.org/r/f49bf0e27eaac396c96d21392c8c284f9f5ef52a.1595543280.git.gustavoars@kernel.org
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/tty/tty_io.c b/drivers/tty/tty_io.c
index 5a6f36b391d9..3b6e5ec3ba54 100644
--- a/drivers/tty/tty_io.c
+++ b/drivers/tty/tty_io.c
@@ -2683,7 +2683,7 @@ struct serial_struct32 {
         compat_int_t    baud_base;
         unsigned short  close_delay;
         char    io_type;
-        char    reserved_char[1];
+        char    reserved_char;
         compat_int_t    hub6;
         unsigned short  closing_wait; /* time to wait before closing */
         unsigned short  closing_wait2; /* no longer used... */
@@ -2691,7 +2691,7 @@ struct serial_struct32 {
         unsigned short  iomem_reg_shift;
         unsigned int    port_high;
      /* compat_ulong_t  iomap_base FIXME */
-        compat_int_t    reserved[1];
+        compat_int_t    reserved;
 };
 
 static int compat_tty_tiocsserial(struct tty_struct *tty,